tests/v2/validation/auth/provider/provider_test.go

```diff
@@ -0,0 +1,484 @@
+		nestedUserPrincipalID := newPrincipalID(authConfigID, "user", v.Username, searchBase)
+
+		userPRTB := &v3.ProjectRoleTemplateBinding{
+			ProjectID:        defaultProject.ID,
+			GroupPrincipalID: nestedUserPrincipalID,
+			RoleTemplateID:   rbac.ProjectOwner.String(),
+		}
+
+		logrus.Infof("Creating project role template binding for user [%v] with role [%v]", userPRTB.GroupPrincipalID, userPRTB.RoleTemplateID)
+		userPRTBResp, err := authAdmin.Management.ProjectRoleTemplateBinding.Create(userPRTB)
+		require.NoError(a.T(), err)
+
+		logrus.Infof("Verifying project role template binding has created for user [%v]", v.Username)
+		assert.NotNilf(a.T(), userPRTBResp, "Error occured while: project role template binding is created for user [%v]", v.Username)
+	}
+
+	var principalIDs []string
+
+	principalIDs = append(principalIDs, newPrincipalID(authConfigID, "group", a.authConfig.DoubleNestedGroup, searchBase))
+	for _, v := range a.authConfig.DoubleNestedUsers {
+		principalIDs = append(principalIDs, newPrincipalID(authConfigID, "user", v.Username, searchBase))
+	}
+
+	logrus.Infof("Verifying access mode updated to restrict access to only the authorized users & groups")
+
+	existing, updates := newWithAccessMode(a.T(), authAdmin, authConfigID, "required", principalIDs)
+	newAuthConfig, err := update(existing, updates)
+
+	require.NoError(a.T(), err)
+	assert.Equal(a.T(), existing.AccessMode, newAuthConfig.AccessMode, "Error occured while: access mode updated to restrict access to only the authorized users & groups")
+
+	for _, v := range a.authConfig.DoubleNestedUsers {
+		user := &v3.User{
+			Username: v.Username,
+			Password: v.Password,
+		}
+		_, err := login(authAdmin, authProvider, user)
+
+		logrus.Infof("Verifying logging as user [%v], should be able to login", v.Username)
+		assert.NoErrorf(a.T(), err, "Error occured while: logging as user [%v], should be able to login", v.Username)
+
+	}
+
+	for _, v := range slices.Concat(a.authConfig.Users, a.authConfig.NestedUsers) {
+		user := &v3.User{
+			Username: v.Username,
+			Password: v.Password,
+		}
+		_, err := login(authAdmin, authProvider, user)
+
+		logrus.Infof("Verifying logging as user [%v], should be able to login", v.Username)
+		assert.Errorf(a.T(), err, "Error occured while: logging as user [%v], should be able to login", v.Username)
+	}
+
+	logrus.Infof("Rolling back the access mode to unrestricted from restrict access to only the authorized users & groups")
+
+	// Rollback only happens with the local client change a.client to authAdmin to reproduce the issue
+	authExisting, authWithUnrestricted := newWithAccessMode(a.T(), a.client, authConfigID, "unrestricted", nil)
+	newAuthConfig, err = update(authExisting, authWithUnrestricted)
+
+	require.NoError(a.T(), err)
+	assert.Equal(a.T(), authExisting.AccessMode, newAuthConfig.AccessMode, "Rolling back the access mode to unrestricted from restrict access to only the authorized users & groups")
```

Comment:
since the message is printed when the assertion fails, can we change it to `Rolling back the access mode to unrestricted from restrict access to only the authorized users & groups `**failed**`"